### ISSUE TITLE
Is it possible to define a global loader function using generouted with react-router?

### ISSUE THREAD

**Issue by atresnjo** [oedotme/generouted#167]
Basically, I want to protect my routes and Loader seems the best way to do it, just wondering is it possible to define the loader once and re-use it across routes? Thanks for the library! Great work 👍 

```ts
export async function Loader() {
	try {
		const data = await trpcClient.authTestRouter.session.fetch()

		const user: User = {
			...data,
		}
		return { user }
	} catch (error) {
		return new Response('Unauthorized', {
			status: 302,
			headers: { location: '/auth/sign-up' },
		})
	}
}
```

**Comment by oedotme** [comment]
@atresnjo Yes, that's possible. You can define this `Loader` as a global `Loader` by adding it to the `src/pages/_app.tsx` layout file similar to the global `Pending` and `Catch` [defined in this react-router example](https://github.com/oedotme/generouted/blob/main/examples/react-router/src/pages/_app.tsx#L5-L9).

If you need a `Loader` at a smaller scope, you can add it to a parent regular layout or a pathless layout group.

Hope that helps!
